policy_network/HiveNet.py

```diff
@@ -38,7 +39,7 @@ def pick_action(self, map_input, thresholds, collector):
     collector.actions.append(action)
     collector.action_logarithms.append(
         distribution.log_prob(action))
-    return action.item()
+    return np.unpackbits(np.uint8(action.item()))[-self.num_of_thresholds:]
```

Comment:
a czemu tak? z czystej ciekawości, bo dla mnie wygląda to czarnomagicznie :D

Reply:
tia... no to troche taka przejściówka, formalnie program mial wybierac z dyskretnej przestrzeni akcji na zasadzie: poprosze akcje nr 9. Nasz program chce dwie akcje dla kazdego thresholda 0 lub 1. Stąd akcje nr 9 rozkladamy na jej reprezentacje binarną i wykorzystujemy k=threshold pierwszych bitów

Reply:
```suggestion
   def bit_representation(action, num_bits):
      return np.unpackbits(np.uint8(action.item()))[-num_bits:]
    return bit_representation(action.item(), num_bits = self.num_of_thresholds)
```

moze byc ewentualnie w ten sposób

Reply:
Czy na pewno chcemy to robić w tym miejscu? Sieć zwraca sobie jedną z 2^k - 1 akcji, gdzie k to liczba thresholdów i przekazujemy taką decyzję do enva. Jakoś tak naturalniej przejście między numerem akcji, a jej interpretacją mi leży w środku metody step enva, niż w środku wybierania akcji przez sieć. Oczywiście jest to rzecz czysto koncepcyjna bo obie wersje koniec końców działają tak samo.

Reply:
Jasne, no ale jako że mamy już mało czasu to może już zostawmy tak jak jest